### PR TITLE
Add eslint, fix eslint issues

### PR DESCRIPTION
**Pull request by pierreTklein**
# Description
 
Migrate from jsHint to esLint for async / await support.

Fix issues that esLint has found.

### CODE REVIEW

File: middlewares/hacker.middleware.js [hunk 1]
```diff
@@ -356,7 +355,7 @@ function checkStatus(statuses) {
 
         let hacker = await Services.Hacker.findById(req.params.id);
 
-        if (!!hacker) {
```
Comment (YiFeiZhang2) [inline middlewares/hacker.middleware.js]: There was an original reason why we did '!!' right?

File: middlewares/team.middleware.js [hunk 2]
```diff
@@ -361,28 +360,6 @@ async function updateHackerTeam(req, res, next) {
...
 
-/**
- * @async
- * @function findById
- * @param {{body: {id: ObjectId}}} req 
- * @param {*} res 
- * @return {JSON} Success or error status
- * @description Finds a team by it's mongoId that's specified in req.param.id in route parameters. The id is moved to req.body.id from req.params.id by validation.
- */
-async function findById(req, res, next) {
```
Comment (YiFeiZhang2) [inline middlewares/team.middleware.js]: Is this being removed because it's being currently unused?

File: services/team.service.js [hunk 3]
```diff
@@ -31,8 +28,6 @@ function findTeamByHackerId(hackerId) {
  * @description Adds a new team to database.
  */
 async function createTeam(teamDetails) {
-    const TAG = `[Team Service # createTeam]:`;
```
Comment (YiFeiZhang2) [inline services/team.service.js]: A lot of unused TAGs, we should discuss whether we want to keep these, and if so, how to utilize them for debugging/logging purposes

File: middlewares/hacker.middleware.js (hunk 1 above)
Comment (pierreTklein) [reply middlewares/hacker.middleware.js]: The original reason was to convert to a Boolean, but eslint considers it unnecessary.

File: middlewares/team.middleware.js (hunk 2 above)
Comment (pierreTklein) [reply middlewares/team.middleware.js]: There is another function definition in this file called `findById`, lol

File: services/team.service.js (hunk 3 above)
Comment (pierreTklein) [reply services/team.service.js]: Yeah, we need to either remove, or use them. Your call.

**Pull request closed without merging**